### ISSUE TITLE
cli: log version details

### ISSUE THREAD

**Issue by stuartc** [OpenFn/kit#130]
### Details

When running cli `execute`, I'd like to see:

- The current version of node
- The adaptor version (both the resolved and specified version)  
  In the case of `@latest` it should show `@1.2.3 [latest]`.
- The version of the CLI at least, and possibly the version of the runtime and/or runtime-manager.

### Implementation notes

Lets try log this out to STDERR, so it's easier to filter in Lightning

**Comment by josephjclark** [comment]
The resolved adaptors should come up in debug logging at least. I think we put something in info? I'll check.

Logging to stderr feels pretty weird and won't play well with how the logger works. Not without effort, anyway. How does it help lightning? Because it's easier to isolate the version output?

We could add a [VER] log level or something which prints version info. That would be easier to filter/detect. You could even suppress all logging apart from VER through the CLI itself

**Comment by stuartc** [comment]
We don't _have_ to log to STDERR - it's just easier to split pipes in childprocess streams when calling out to the shell.

Like if it's gonna break a bunch of stuff we can leave it. But if it's doable it would be nice, regular shell users won't notice anything.

**Comment by josephjclark** [comment]
I've had a little swing at this in #138 but I'm really not that happy about it.

You can now do `openfn version`, which will print out version diagnostic info. It could also include adaptor version if you pass an argument.

Would it make more sense to run `openfn version` first, divert that output to wherever Lightning needs it, and then run the actual job?

**Comment by taylordowns2000** [comment]
@josephjclark , @stuartc , not sure of _how_ (of course... sorry!) but the goal here is to have the adaptor version show up inside this BEAUTIFUL list:
![image.png](https://images.zenhubusercontent.com/566ac5e0638e718c995f5356/d4c548aa-d117-43ff-b6a4-3369b1e28f93)

**Comment by stuartc** [comment]
@josephjclark this output is usually used in debugging on production, like knowing exactly what version the job was executed with, since `latest` doesn't mean much a week after the event - and whenever we were on occasion fixing things with core, it was good to know that a given machine in the cluster was in fact up to date etc.

So these kinds of details are pretty important, if you feel they are noisy when using it via the terminal we can perhaps come up with a way to make them opt-in or tagged in a way they can be filtered out.. Maybe a flag is the way forward for that.

To summarise, we want to know the adaptor version as well. In core after we resolve the adaptor, we resolve the 'root' (this may actually be very difficult with ESM modules :$ ) and read the package.json file to get the specific version

**Comment by josephjclark** [comment]
Yep, it's coming.

It's a bit complicated because of how adaptors are loaded - we actually do a lot of logging while loading adaptors and that all needs pre-empting to put it up at the top like this. There's a PR open (#136) which sort of does it but I'm not happy with the solution. It'll get sorted in the next couple of days.

@stuartc there's a couple of things here I could use your input on (especially the stderr stuff) so maybe we can have a chat tomorrow/Friday?

**Comment by josephjclark** [comment]
Quick update after talking with Stu.

The thing with stderr is that MIGHT want a "pure output, no filesystem" mode, where all logging gets output to stderr, and only the final state  gets output to stderr. This lets you pipe there result into other commands while still retaining logging via the stderr stream.

I am all for doing that at some stage but it's a global logging change, not actually related to versioning.

So I'm gonna wrap up this work now and remove stderr as a requirement.
